webpack.config.js

```diff
@@ -0,0 +1,234 @@
+    loader: 'url-loader',
+    query: { name: '[hash].[ext]', limit }, // name for file-loader
+    include: PATHS.app
+  });
+
+  const babelLoader = (production = false, browser = false) => {
+    const obj = {
+      test: /\.js$|\.jsx$/,
+      loader: 'babel-loader',
+      query: {
+        presets: ['es2015', 'react', 'stage-0'],
+        plugins: ['transform-decorators-legacy']
+      },
+      exclude: PATHS.modules
+    };
+    if (production) {
+      obj.query.plugins.push(
+        'transform-react-remove-prop-types',
+        'transform-react-constant-elements',
+        'transform-react-inline-elements'
+      );
+    } else if (browser) {
+      obj.query.presets.unshift('react-hmre');
+    }
+    return obj;
+  };
+
+  const cssLoader = (production = false, browser = false) => {
+    /*
+     * https://github.com/webpack/css-loader/issues/59 - use css-loader/locals on server-side rendering
+     * :local(.className) used to declare className in the local scope,
+     * local identifiers are exported by css-loader
+     *
+     * modules enable CSS Modules spec https://github.com/css-modules/css-modules
+     *
+     * importLoaders (int): That many loaders after the css-loader are applied to @import-ed resources.
+     *
+     */
+    const queryModules = 'modules';
+    const queryImportLoaders = 'importLoaders=1';
+    const queryLocal = 'localIdentName=[name]__[local]___[hash:base64:5]';
+
+    const queryStrExtractTextPlugin = '?' + queryModules + '&' + queryImportLoaders;
+    const queryStr = queryStrExtractTextPlugin + '&' + queryLocal;
+
+    const serverSide = 'css-loader/locals' + queryStr + '!postcss-loader';
+    const browserSide = 'style-loader!css-loader' + queryStr + '!postcss-loader';
+
+    const myLoader = browser ? browserSide : serverSide;
+
+    const prodBrowserRenderLoader = ExtractTextPlugin.extract({
+      fallbackLoader: 'style-loader',
+      loader: 'css-loader' + queryStrExtractTextPlugin + '!postcss-loader'
+    });
+    const loader = production && browser ? prodBrowserRenderLoader : myLoader;
+    const obj = { test: /\.css$/, include: PATHS.app, loader };
+    return obj;
+  };
+
+  const webpackPlugins = (production = false, browser = false) => {
+    const bannerOptions = { raw: true, banner: 'require("source-map-support").install();' };
+    const compress = { warnings: false };
+
+    if (!production && !browser) {
+      return [
+        new webpack.EnvironmentPlugin(['NODE_ENV']),
+        new webpack.IgnorePlugin(/vertx/),
+        new webpack.BannerPlugin(bannerOptions)
+      ];
+    }
+    if (!production && browser) {
+      return [
+        new webpack.HotModuleReplacementPlugin(),
+        new webpack.NoErrorsPlugin(),
+        new webpack.EnvironmentPlugin(['NODE_ENV'])
+        // stylelint-webpack-plugin does not support webpack 2 yet
+        // new styleLintPlugin({
+        //   configFile: path.resolve(__DIR__, '.stylelintrc'),
```

Comment:
Has anyone found stylelint-webpack-plugin useful for us? Or should we just 🔪 it?